### ISSUE TITLE
Timeout on second request

### ISSUE THREAD

**Issue by mattboardman** [datdenkikniet/ipmi-rs#1]
I'm using a dell r610 with ipmi 2.0 support. First off, the library doesn't work in a few areas because the rmcp protocol specified all fields need to be big endian encoded. IPMI spec says fields need to be little endian. 

I'm not super familiar with the session management which is why I appreciated your library handling that for me. However, I'm not sure if the session is working fully either. Once I'm able to establish a session (had to tweak the endianess) I can only send 1 command. When I send the next command the rmcp endpoint times out. I assume this is session related since that would be the only thing that would change between requests. Would definitely appreciate some guidance here. 

Do you have a link to the spec docs you used when writing this library?

**Comment by mattboardman** [comment]
I actually figured it out. 

According to the spec until a session is established the sequence number should be set to 0. In the library [here] (https://github.com/datdenkikniet/ipmi-rs/blob/9bc9f9e4fb711fe37feaeb38ce321a9d7d7cf172/src/connection/impls/rmcp/mod.rs#L300), you increment the sequence number on every request even before a session is established like [here](https://github.com/datdenkikniet/ipmi-rs/blob/9bc9f9e4fb711fe37feaeb38ce321a9d7d7cf172/src/connection/impls/rmcp/mod.rs#L217) and then [here](https://github.com/datdenkikniet/ipmi-rs/blob/9bc9f9e4fb711fe37feaeb38ce321a9d7d7cf172/src/connection/impls/rmcp/mod.rs#L229). 

A simple solution was to increment the sequence number if the session_id was no longer 0. Something simple like this will work (mind the sloppiness).

```
let session_sequence = match self.state.session_id {
  0 => self.state.request_sequence,
  _ => {
    // we want to increment first because when we established 
    // the session id during the challenge that's the initial sequence number of 0
    self.state.request_sequence = self.state.request_sequence.wrapping_add(1);
    self.state.request_sequence
  }
};
```

**Comment by datdenkikniet** [comment]
I have now merged #3 which should address this issue. Please re-open if that is not the case :)
